client/src/components/CheckoutOptions.jsx

```diff
@@ -1,20 +1,69 @@
+/* eslint-disable arrow-body-style */
+/* eslint-disable no-useless-constructor */
+/* eslint-disable class-methods-use-this */
 import React from 'react';
 
 
 class CheckoutOptions extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      colorModifier: 1,
+      quantity: 1,
+    };
+
+    this.handleColorChoice = this.handleColorChoice.bind(this);
```

Comment:
Did you do this just so you didn't have to write 'this' every time  you call the function? Interesting!

Reply:
Its just become like a habit for me to add that anytime i'm going to call a function from an onclick or onchange

Reply:
Basically since color choice has to take in an event, we need to bind that event to the function when its called so since i'm calling this.handleColorChoice in an onChange, It has to have bind on it. I could have just written the onchange like this:
`onChange={this.handleColorChoice.bind(this)}`
But I would rather have it so that when I call this.handleColorChoice it is always binding

Reply:
ohhhh okay I gotchu